lib/utils/ensure-single-browser-tab-only.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import BootWarning from '../components/boot-warning';
+
+const HEARTBEAT_DELAY = 1000;
+const clientId = uuidv4();
+console.log(clientId);
+const emptyLock = [null, -Infinity];
+const foundElectron = window.process && window.process.type;
+
+if (!foundElectron) {
+  const sesh = grabSessionLock();
+  console.log(sesh);
+  if ('lock-acquired' !== sesh) {
+    ReactDOM.render(<BootWarning />, document.getElementById('root'));
+    throw new Error('Simplenote can only be opened in one tab');
+  }
+  let keepGoing = true;
+  loop(() => {
+    if (!keepGoing) {
+      return false;
+    }
+    switch (grabSessionLock()) {
+      case 'lock-acquired':
+        return true; // keep updating the lock and look for other sessions which may have taken it
+
+      default:
+        window.location.reload();
```

Comment:
I'm afraid that this will cause unexpected refreshes and maybe a refresh cycle.

at this point we have booted the app and acquired the lock at least once. we never expect this scenario but it could happen in at least a few rare circumstances:
 - our app slows down. maybe a very large not takes longer to update and delays the lock update. another session could have stolen the lock in that delay window.
 - someone directly messes with `localStorage` in the browser console.

should these things happen I don't know how best to respond but at this point we're taking the nuclear option and reloading everything, possibly interrupting someone who is working on a note and without alerting them to why.

the `window.alert()` was an attempt at putting up a modal they couldn't ignore that warns them of their predicament and gives them a choice to reload or not; to wait a second and see if they are willing to risk the bug in order to try and save what they had open.

thoughts?